src/components/ContactLocations/index.jsx

```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+import './styles.css';
+import { useCharityAPI } from '../../clients';
+import MyLoader from './loader.jsx';
+
+export default function ContactLocations() {
+  const { data, loading, dataError } = useCharityAPI('/branch-contacts');
+
+  if (dataError) {
+    return <div>Can Not Get Data!</div>;
+  }
+
+  if (loading) {
+    return (
+      <section className="container">
+        <div className="address grid lg:grid-cols-3 gap-8 sm:grid-cols-1 md:grid-cols-2 justify-between box-border">
+          <MyLoader />
+          <MyLoader />
+          <MyLoader />
+        </div>
+      </section>
+    );
+  }
+  if (data) {
+    return (
+      <section className="container">
+        <div className="address grid lg:grid-cols-3 gap-8 sm:grid-cols-1 md:grid-cols-2 justify-between box-border">
+          {data.map(({ city, email, location, phone, id }) => (
+            <Location
+              city={city}
+              email={email}
+              location={location}
+              phone={phone}
+              key={id}
+            />
+          ))}
+        </div>
+      </section>
+    );
+  }
+  return <div>Generic Error!</div>;
+}
+
+const Location = ({ city, location, email, phone }) => {
+  return (
+    <div className="container__location p-10">
+      <h6 className="location-city text-lg pl-2 text-c000">{city}</h6>
+
+      <div className="flex pt-8 items-center">
+        <span className="flex-shrink-0">
+          <img
+            className="contact-location-icons object-fill object-cover"
+            src={location.icon.url}
```

Comment:
destructure at the top of the function

Reply:
I think destructuring ever object of the response to get the specific properties inside the nested objects will end up with a big confusion.